internal/utils/poller/poller.go

```diff
@@ -4,17 +4,16 @@ import (
 	"context"
 	"time"
 
-	"github.com/babylonlabs-io/babylon-staking-indexer/internal/types"
 	"github.com/rs/zerolog/log"
 )
 
 type Poller struct {
 	interval   time.Duration
 	quit       chan struct{}
-	pollMethod func(ctx context.Context) *types.Error
+	pollMethod func(ctx context.Context) error
 }
 
-func NewPoller(interval time.Duration, pollMethod func(ctx context.Context) *types.Error) *Poller {
+func NewPoller(interval time.Duration, pollMethod func(ctx context.Context) error) *Poller {
```

Comment:
return value is used only in this case `log.Error().Err(err).Msg("Error polling")` which expects `error` anyway